src/main/java/com/flipkart/ranger/healthservice/HealthService.java

```diff
@@ -0,0 +1,54 @@
+package com.flipkart.ranger.healthservice;
+
+import com.flipkart.ranger.healthcheck.HealthcheckStatus;
+import com.flipkart.ranger.healthservice.monitor.IsolatedHealthMonitor;
+import com.flipkart.ranger.healthservice.monitor.Monitor;
+
+/**
+ * An interface which can be used to track the health of any service
+ * You can add a set of monitors, and a general health will be tracked by
+ *
+ * @see ServiceHealthAggregator
+ */
+public interface HealthService {
+
+    /**
+     * Register a monitor to the service, to setup a continuous monitoring on the monitor
+     * <p/>
+     * this method can be used to add a {@link IsolatedHealthMonitor} which will later be
+     * scheduled at regular intervals and monitored to generate and maintain an aggregated health of the service
+     * the scheduling will happen in an isolated thread
+     *
+     * @param monitor an implementation of the {@link IsolatedHealthMonitor}
+     */
+    void addIsolatedMonitor(IsolatedHealthMonitor monitor);
```

Comment:
We have differentiated two types of monitors consciously as I don't want many threads running independently on my service. There are few usecases where a monitor might need to run in it's own state (I have an usecase right now at hand on something I'm working on). However, cpu-mon, mem-mon- etc can run in series. Not everything needs to run in parallel and keep accumulating data, some stuff is needed real-time. For example DB connectivity. As such, we'd like to atleast keep the provision for monitors that are not isolated. I believe it's critical for people to understand this behavioral difference when registering a monitor. The naming therefore helps the user to clearly understand this differentiation.